extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js

```diff
@@ -0,0 +1,343 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Detects <amp-img> elements to set the `lightbox` attribute
+ * automatically.
+ *
+ * This extension is not meant to be included explicitly, e.g. as a script tag.
+ * Instead, the runtime loads it when encountering an <amp-img>.
+ */
+
+import {AmpEvents} from '../../../src/amp-events';
+import {CommonSignals} from '../../../src/common-signals';
+import {Services} from '../../../src/services';
+import {closestBySelector} from '../../../src/dom';
+import {dev} from '../../../src/log';
+import {getMode} from '../../../src/mode';
+import {toArray} from '../../../src/types';
+import {tryParseJson} from '../../../src/json';
+
+
+const TAG = 'amp-auto-lightbox';
+
+export const REQUIRED_EXTENSION = 'amp-lightbox-gallery';
+export const LIGHTBOXABLE_ATTR = 'lightbox';
+
+export const VISITED_ATTR = 'data-amp-auto-lightbox-visited';
+
+export const ENABLED_SCHEMA_TYPES = [
+  'Article',
+  'NewsArticle',
+  'BlogPosting',
+  'LiveBlogPosting',
+  'DiscussionForumPosting',
+];
+
+/** Factor of naturalArea vs renderArea to lightbox. */
+export const RENDER_AREA_RATIO = 1.2;
+
+/** Factor of renderArea vs viewportArea to lightbox. */
+export const VIEWPORT_AREA_RATIO = 0.3;
+
+const ACTIONABLE_ANCESTORS =
+    'a[href], amp-selector > [option], amp-script, amp-story';
```

Comment:
`option` does not need to be direct child of `amp-selector`, so `amp-selector [option]` (my bad from previous comment)